timemachine/cpp/src/kernels/k_nonbonded.cuh

```diff
@@ -345,29 +379,33 @@ void __global__ k_nonbonded_du_dx(
                 RealType sig6_inv_d8ij = sig6_inv_d6ij*inv_d2ij;
                 delta_prefactor -= eps_ij*sig6_inv_d8ij*(sig6_inv_d6ij*48 - 24);
             }
-            
-            // RealType delta_prefactor = es_prefactor-lj_prefactor;
 
             gi_x += FLOAT_TO_FIXED(delta_prefactor*delta_x);
             gi_y += FLOAT_TO_FIXED(delta_prefactor*delta_y);
             gi_z += FLOAT_TO_FIXED(delta_prefactor*delta_z);
 
-            gj_x -= FLOAT_TO_FIXED(delta_prefactor*delta_x);
-            gj_y -= FLOAT_TO_FIXED(delta_prefactor*delta_y);
-            gj_z -= FLOAT_TO_FIXED(delta_prefactor*delta_z);
+            // (ytz): the cast relies on undefined behavior since the exclusion terms will overflow.
+            // Thus far, the current gen GPUs are deterministic in the resulting value. So long
+            // as the UB stays deterministic then this should be fine. There's no guarantee though that
+            // NVIDIA will keep the behavior consistent. Furthermore, do *not* attempt to swap signs via:
+            // gj_* -= FLOAT_TO_FIXED(delta_prefactor*delta_*); This is incorrect
```

Comment:
This still makes me a bit nervous.

Perhaps after this PR we can discuss offline if there are any other options for handling exclusions that could be nearly as fast as this approach

Reply:
Agreed - there's other assumptions we make in the code that's somewhat similar. Eg. we assume that the warp size is always 32 threads though NVIDIA doesn't guarantee this.